### DISCUSSION TITLE
無料でワークロードをホストできるサービス Vercel の紹介

### DISCUSSION THREAD

**Discussion by sheep-snow** [nekono-dev/skyshare#39] [Ideas]
ローコストなサービスでワークロードを維持する工夫をされていましたので、Cloudflareに加え、VercelのHobbyプランをastroのワークロードをホストする用途に併用して頂くことがお役に立つかもしれないと思い紹介させて頂きます。

なお、後述の手順で、既存のコードを直接変更せず自動変換に任せるだけでもVercelへのデプロイが成功できましたので、移行のご負担も低いのではないかと思います。

* Vercelがどのように使えるかわかりやすく解説された記事: https://note.com/libproc/n/n10dd8760d530
* 現行Astroのコードのデプロイ先をVercelに変えてデプロイする手順: https://docs.astro.build/ja/guides/deploy/vercel/
* 手順を手元でやってみた例: https://github.com/sheep-snow/skyshare/pull/3/files
* Hobby Plan の制限: https://vercel.com/docs/limits/overview#general-limits
  * 解説 https://dev-harry-next.com/infrastructure/vercel-detail

上記の手順でastroをVercelにデプロイした状態
![image](https://github.com/nkte8/skyshare/assets/71830342/143289f8-8027-403a-96d7-f12909225ac9)

**Comment by nekono-dev** [comment]
今って純正のNodejs上で動いてないから、この辺が改善されることで開発がやりやすくなる可能性はあるかもしれない...
一方、詳細をみるとやっぱりCloudflareよりは少し条件厳しめ（月間のデプロイ可能数がCloudflareが500なのがVercalだと200など）なので、どこまで許容できるか、というところは課題になりそうです。
それでもなにより、選択肢が増える分にはよきです！ありがとうございます！

**Reply by sheep-snow** [reply]
みて下さってありがとうございます！デプロイが活発なのでデプロイ可能数は大切でしたね…考慮できていませんでした。教えていただいてありがとうございます🙏
